crates/conjure_core/src/rule_engine/rewrite.rs

```diff
@@ -134,14 +134,17 @@ pub fn rewrite_model<'a>(
     let start = std::time::Instant::now();
 
     //the while loop is exited when None is returned implying the sub-expression is clean
-    while let Some(step) = rewrite_iteration(
-        &new_model.constraints,
-        &new_model,
-        &rules,
-        apply_optimizations,
-        &mut stats,
-    ) {
-        step.apply(&mut new_model); // Apply side-effects (e.g. symbol table updates)
+    let constraints = new_model.constraints.clone();
```

Comment:
fine for now, but just to note that in gen_reduce (or whatever it's called) this for loop shouldn't be necessary. it's a container containing expressions, the top level container can be a vector of expressions or just an expression, it shouldn't matter via uniplate/biplate. fyi @lixitrixi @niklasdewally @YehorBoiar

Reply:
So, we can just use model.constraints.transform instead?

Reply:
I think it should be a descend not a transform

Reply:
I think Oz wants `for expr in <_ as Biplate<Expr>>::children(constraints)>`

but what you have now is fine

Reply:
It would be a `<_ as Biplate<Expr>>::descend_bi(constraints)` i think.

If you only want to operate on your immediate children, use descend not transform.

Reply:
I have a mutable iterator in the roadmap which also work nicely here, but for now descend_bi is the better option.